plasma_framework/docs/design/add-new-tx-type-check-lists.md

```diff
@@ -0,0 +1,9 @@
+# Check List For Adding New Transaction Types
+
+This document provides some check lists that whenever we are designing a new transaction type. Since there are some known restriction of the Plasma Framework design, please make sure that new tx type and exit game does not break the global assumptions and limitations.
+
+1. Aside from deposit transactions, all transactions should be unique. This promises the `OutputId` to be unique as `OutputId` uniqueness is a global assumption placed to `PlasmaFramework`.
+1. output index is max to `TX_OFFSET`  (10000) of the `UtxoPos` schema. As a result, no more than such amounts of inputs or outputs in one transaction.
+1. `ExitId` for the new exit game contract must promise to be not collide with all existing `ExitId` schemas used in another exit game contracts. (Each exit game contract can potentially introduce new schema if needed. Not recommended though)
+1. Please always define new transaction type and new output type. No reusing on the output type. For example, even an extension of Payment V1 to Payment V2, the output should be using different output type. (see previous discussion: [here](https://github.com/omisego/plasma-contracts/issues/214#issuecomment-526466192))
```

Comment:
When defining a new tx type, a new output type should be introduced along. Do not reuse an already existing output type.